frontend/src/metabase/components/ExplicitSize/ExplicitSize.tsx

```diff
@@ -47,12 +47,15 @@ type InnerProps = {
 
 type ExplicitSizeOuterProps<T> = Omit<T, "width" | "height">;
 
+/**
+ * @deprecated HOCs are deprecated
```

Comment:
Did we decide to deprecate them? If so is there any documentation on why, I can't find anything on notion.

AFAICT React has not deprecated them and is actively pushing them as an alternative to mixins still (even though they are less needed for most scenarios because of hooks).

Reply:
Apparently they're deprecated since before I joined:
- https://github.com/metabase/metabase/pull/30986#issuecomment-1563061794
- https://github.com/metabase/metabase/pull/31059

> If so is there any documentation on why, I can't find anything on notion.

I'm unaware.

Reply:
If so that's strange. HOC components (while slightly unwieldy sometimes) are common patterns in React. Don't let this block the PR though.